docker/sagemaker/Dockerfile.sm

```diff
@@ -1,22 +1,11 @@
 # Docker file for building a docker image for running GraphStorm code on Amazon SageMaker
-# Note: Distributed graph partition will use another docker image which will come soon.
-
 ARG DEVICE=gpu
-ARG DGL_VERSION=2.3.0
+ARG SOURCE
+ARG USE_PARMETIS=false
 
-FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker as branch-gpu
```

Comment:
It may affect our sagemaker regression CI as we rely on our current docker file to first build an image and then start the job. I would suggest we can make the change at the same time.

Reply:
Could you clarify what would break here? Updating the build script along with the Dockerfiles in this PR should allow us to build the image?

Reply:
I'll see if I can revert these changes, and keep the breaking change for v0.5, at which point we can deprecate the old build scripts.